### PR TITLE
Add supporter status to the supporter page

### LINKED ISSUE #2027: [Old-Feature] Missing osu!supporter expiration date check in new web
Old supporter page: https://osu.ppy.sh/p/support
New supporter page: https://osu.ppy.sh/home/support

Suppose I have an active osu!supporter, **where** do I refer to the **expiration date for the osu!supporter** in the **new web**?

In the old web, I can just visit the old supporter page and the expiration date is shown at the top.
For the new web, I have no idea where to find it.

Issue is related to wiki page for osu!supporter: https://github.com/ppy/osu-wiki/pull/706

### LINKED ISSUE #2782: Remaining time of supporting missing from Supporter page
On the old osu! website when you click the supporter option it comes up with how long you have left of your supporter tag, But on the new website this feature is not there.

### PR DESCRIPTION
**Pull request by nekodex**
The supporter page might end up being merged with the store/purchase page later, but for the time being the status can go here.

fixes #2782, fixes #2027

### CODE REVIEW

**Comment by MegaApplePi** [comment]
also resolves #2027 ... #2782 was a duplicate :man_shrugging:

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -165,35 +166,80 @@ public function setLocale()
...
-            ],
+        if (Auth::check()) {
+            $user = Auth::user();
+
+            // current status
+            $expiration = $user->osu_subscriptionexpiry;
+            $current = $expiration !== null ? $expiration->isFuture() : false;
+
+            // purchased
+            $tagPurchases = $user->supporterTagPurchases()->get();
```
[the remark this answers is not in the document]
Comment (notbakaneko) [reply app/Http/Controllers/HomeController.php]: Would depend if you want to use the methods defined by `Collection` to filter through the existing collection as opposed to running more queries when called through `Builder`

File: resources/views/home/support-the-game.blade.php [hunk 2]
```diff
@@ -54,8 +54,54 @@
     </div>
 
     <div class="osu-page osu-page--small osu-page--stg-block">
+        @if (!empty($supporterStatus))
+        <!-- supporter status  -->
+        <div class="stg-status{{ $supporterStatus['current'] ? ' stg-status--active' : '' }} stg-block__top">
```
[the remark this answers is not in the document]
Comment (nekodex) [reply resources/views/home/support-the-game.blade.php]: which? the theme modifier?

File: app/Http/Controllers/HomeController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (nekodex) [reply app/Http/Controllers/HomeController.php]: what notbakaneko said 👍

File: resources/views/home/support-the-game.blade.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (nanaya) [reply resources/views/home/support-the-game.blade.php]: `stg-block__top`

File: app/Http/Controllers/HomeController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (nanaya) [reply app/Http/Controllers/HomeController.php]: but `$user->supporterTagPurchases()->get()` and `$user->supporterTagPurchases` are same thing...?

[the remark this answers is not in the document]
Comment (nanaya) [reply app/Http/Controllers/HomeController.php]: (note the missing `()`)

**Pull request merged**